File: fixtures/struct-roundtrip.swift
struct User {
  var id: String
  var age: Int32
  var nickname: String?
  var tags: [String]
}

func birthday(user: User) throws -> User {
  return User(id: user.id, age: (user.age &+ 1), nickname: user.nickname, tags: user.tags)
}

func describe(user: User) throws -> String {
  let nickname: String? = user.nickname
  if nickname == nil {
    return user.id + " (" + lucentStr(user.age) + ")"
  }
  return nickname! + " aka " + user.id
}

File: fixtures/struct-roundtrip.kt
data class User(
  var id: String,
  var age: Int,
  var nickname: String?,
  var tags: MutableList<String>
)

fun birthday(user: User): User {
  return User(id = user.id, age = user.age + 1, nickname = user.nickname, tags = user.tags)
}

fun describe(user: User): String {
  val nickname: String? = user.nickname
  if (nickname == null) {
    return user.id + " (" + lucentStr(user.age) + ")"
  }
  return nickname!! + " aka " + user.id
}
